From 3961201353cc6761366ed19728ecad19e1467aeb Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Mon, 19 Mar 2012 10:52:34 +0800
Subject: [PATCH 284/435] ENGR00177123: Add check for OBJDIR when not use LTIB
 build

1. Fixed the build break when use `make -C test/doc/ PLATFORM=IMX6Q'

> asciidoc -a toc ut_um.toc
> cp -f *.html
> cp: missing destination file operand after `ut_um.html'
> Try `cp --help' for more information.
> make: *** [chunked] Error 1

2. Also moved the config generating from Makefile to a newly-created script.

Signed-off-by: Robby Cai <R63905@freescale.com>
---
 test/doc/Makefile    | 14 +++++++++-----
 test/doc/gen_conf.sh |  8 ++++++++
 2 files changed, 17 insertions(+), 5 deletions(-)
 create mode 100755 test/doc/gen_conf.sh

diff --git a/test/doc/Makefile b/test/doc/Makefile
index df65a68..15b4ac6 100644
--- a/test/doc/Makefile
+++ b/test/doc/Makefile
@@ -1,19 +1,23 @@
 DOCMAIN :=$(wildcard *.toc)
-PLATFORMLOWER := $(shell echo $(PLATFORM) | tr A-Z a-z)
 
-all: chunked single
+all: pre-build chunked single
+
+pre-build:
+	./gen_conf.sh $(PLATFORM)
 
 chunked: $(DOCMAIN)
-	echo "// Don't edit it manually!" > def.cfg
-	echo ":chip: $(PLATFORM)" >> def.cfg
-	echo ":$(PLATFORMLOWER):" >> def.cfg
 	asciidoc -a toc $<
+ifneq ($(OBJDIR),)
 	cp -f *.html $(OBJDIR)
+endif
+
 single:
 
 pdf: $(DOCMAIN)
 	a2x -dbook -f pdf $<
+ifneq ($(OBJDIR),)
 	cp -f *.pdf $(OBJDIR)
+endif
 
 .PHONY: clean
 clean:
diff --git a/test/doc/gen_conf.sh b/test/doc/gen_conf.sh
new file mode 100755
index 0000000..659e526
--- /dev/null
+++ b/test/doc/gen_conf.sh
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+PLATFORM=$1
+PLATFORMLOWER=`echo ${PLATFORM} | tr A-Z a-z`
+
+echo "// Don't edit it manually!" > def.cfg
+echo ":chip: ${PLATFORM}" >> def.cfg
+echo ":${PLATFORMLOWER}:" >> def.cfg
-- 
1.8.0

